includes/embeds/class-amp-tiktok-embed-handler.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ *
+ * @package AMP
+ */
+
+use Amp\AmpWP\Dom\Document;
+
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ */
+class AMP_Tiktok_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Registers embed.
+	 */
+	public function register_embed() {
+		// Not implemented.
+	}
+
+	/**
+	 * Unregisters embed.
+	 */
+	public function unregister_embed() {
+		// Not implemented.
+	}
+
+	/**
+	 * Sanitize Tiktok embeds to be AMP compatible.
+	 *
+	 * @param Document $dom DOM.
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+		$nodes = $dom->xpath->query( '//blockquote[ @class="tiktok-embed" ]' );
+
+		foreach ( $nodes as $node ) {
+			if ( ! $this->is_raw_embed( $node ) ) {
+				continue;
+			}
+
+			$this->make_embed_amp_compatible( $node );
+		}
+	}
+
+	/**
+	 * Determine if the node has already been sanitized.
+	 *
+	 * @param DOMElement $node The DOMNode.
+	 * @return bool Whether the node is a raw embed.
+	 */
+	protected function is_raw_embed( DOMElement $node ) {
+		return ! $node->firstChild || ( $node->firstChild && 'amp-iframe' !== $node->firstChild->nodeName );
+	}
+
+	/**
+	 * Make Tiktok embed AMP compatible.
+	 *
+	 * @param DOMElement $node The DOMNode to make AMP compatible.
+	 */
+	protected function make_embed_amp_compatible( DOMElement $node ) {
+		$dom       = $node->ownerDocument;
+		$embed_url = $node->getAttribute( 'cite' );
+		$video_id  = $node->getAttribute( 'data-video-id' );
+
+		$this->remove_embed_script( $node );
+
+		// Create fallback node.
+		$fallback_node = AMP_DOM_Utils::create_node(
+			$dom,
+			'a',
+			[
+				'href'  => esc_url_raw( $embed_url ),
+				'class' => 'amp-wp-embed-fallback',
+			]
+		);
+		$fallback_node->appendChild( new DOMText( esc_html( $embed_url ) ) );
+
+		// If there is no video ID, replace the blockquote with the fallback and return.
+		if ( empty( $video_id ) ) {
+			$node->parentNode->replaceChild( $fallback_node, $node );
```

Comment:
I think rather than replace with a fallback, this method should short-circuit without doing anything.